src/libs/Tiktoken.Core/CoreBPE.cs

```diff
@@ -89,23 +98,40 @@ public int CountTokensNative(string text)
         var textSpan = text.AsSpan();
         Span<byte> pieceBytes = stackalloc byte[128];
 #endif
+#if NET9_0_OR_GREATER
+        var fastEncoderLookup = FastEncoder.GetAlternateLookup<ReadOnlySpan<char>>();
```

Comment:
Here you use FastEncoder in addition to FastCacheCounts, but you didn't pass new AlternateStringComparer() to it.
Is this implemented by default for regular Dictionary and is only needed for ConcurrentDictionary?
P.S. I'm a bit out of context for now, I'll look into this in the next few days (I saw links on your blog, I'll look into this)